### ISSUE TITLE
Gallery view focus not updated correctly when deleting files

### ISSUE THREAD

**Issue by ChloeSartorelli** [silverstripe/silverstripe-asset-admin#1670]
As a keyboard user, when you have focused on a file in the gallery view, select it, and delete it, your focus should be moved to the item that is _now_ in the position that the deleted file _was_ in.

This works as expected if you're on the last page, but if you're _not_ on the last page, it doesn't work.

This is because the lifecycle event that triggers checks if we have less items than before - but only checks if the number of items on the _current page_ has changed.
When you delete an item on the current page, it will pull an item from the next page into the current page, so the number of items per page doesn't change - so the focus change doesn't get triggered.

## Possible solutions

1. Track the total number of items rather than the number of items per page, or
2. Check for _any changes_ in the IDs and Queued IDs of all items on the current page

Obviously option 1 is the better option here.
